### ISSUE TITLE
Questions about `UnionLit` documentation

### ISSUE THREAD

**Issue by joneshf** [dhall-lang/dhall-haskell#359]
I noticed that in the documentation for `UnionLit` shows it using a tuple.

https://github.com/dhall-lang/dhall-haskell/blob/ac68adc14e9c3d4a870e2887c3f61c68b87b562c/src/Dhall/Core.hs#L316

However, that's not really how you'd create one. It's a bit confusing since most of the other cases are how you'd create one. Is that intentional or a typo? If it's a typo, I'll make a PR. If it's intentional, should we change the `UnionLit` case to be `| UnionLit (Text, Expr s a) (InsOrdHashMap Text (Expr s a))` so it matches the documentation?

In that line, the `t1` should probably be `v1` right? This is a **v**alue, not a **t**ype. But also, there's one blessed key and value and some enumeration of keys and types. Does it make sense to instead say:

```Haskell
-- | > UnionLit (k, v) [(k1, t1), (k2, t2)]   ~  < k = v | k1 : t1 | k2 : t2 >
```
or
```Haskell
-- | > UnionLit k v [(k1, t1), (k2, t2)]   ~  < k = v | k1 : t1 | k2 : t2 >
```
?

**Comment by Gabriel439** [comment]
Yeah, that's just a typo in the documentation.  It should be `UnionLit k v ...` and not `UnionLit (k, v)`.  I think I might have initially written that documentation line back in a first draft where it used a tuple and then possibly changed the constructor without fixing the documentation.

**Comment by joneshf** [comment]
Fixed in #360
